segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -641,7 +643,30 @@ public void truncateSegment(TruncateSegment truncateSegment) {
         log.info(truncateSegment.getRequestId(), "Truncating segment {} at offset {}.",
                 segment, offset);
         segmentStore.truncateStreamSegment(segment, offset, TIMEOUT)
-                .thenAccept(v -> connection.send(new SegmentTruncated(truncateSegment.getRequestId(), segment)))
+                .whenComplete((properties, exp) -> {
+                    if (exp != null) {
```

Comment:
I think we can keep it as is and handle truncation of index segment once main segment get truncated successfully. Its okay if main segment truncation is successful but failed to truncate index segment. @anju-c-das @munish1789

Reply:
yes instead of introducing when complete here, a private method  truncateIndexSegment can be created with functionality and called after main segment truncation is successful.

Reply:
something like 
`private void truncateIndexSegment(String segment, long offset) {
        long indexSegmentOffset = IndexRequestProcessor.locateOffsetForSegment(segmentStore, segment, offset, true);
        segmentStore.truncateStreamSegment(getIndexSegmentName(segment), indexSegmentOffset, TIMEOUT);
    }`

Reply:
made the changes as suggested.